src/senaite/core/upgrade/v02_06_000.py

```diff
@@ -1731,6 +1734,162 @@ def move_interpretationtemplates(tool):
+
+    # run required import steps
+    tool.runImportStepFromProfile(profile, "typeinfo")
+    tool.runImportStepFromProfile(profile, "workflow")
+
+    origin = api.get_setup().get("bika_suppliers")
+    destination = get_setup_folder("suppliers")
+
+    # un-catalog the old container
+    uncatalog_object(origin)
+
+    query = {"portal_type": "Supplier"}
+    # search all AT based suppliers
+    brains = api.search(query, SETUP_CATALOG)
+    total = len(brains)
+
+    # get all objects first
+    objects = map(api.get_object, brains)
+    for num, obj in enumerate(objects):
+        migrate_supplier_to_dx(obj, destination)
+
+        logger.info("Migrated supplier {0}/{1}: {2} -> {3}".format(
+            num, total, api.get_path(obj), api.get_path(obj)))
+
+    if origin:
+        # remove old AT folder
+        if len(origin) == 0:
+            delete_object(origin)
+        else:
+            logger.warn("Cannot remove {}. Is not empty".format(origin))
+
+    logger.info("Convert Suppliers to Dexterity [DONE]")
+
+
+def migrate_supplier_to_dx(src_supplier, destination):
+    """Migrates a Supplier to DX in destination folder
+
+    :param src_supplier: The source AT object
+    :param destination: The destination folder
+    """
+
+    # Create the object if it does not exist yet
+    src_id = src_supplier.getId()
+    target_id = src_id
+
+    target = destination.get(target_id)
+    if not target:
+        # Don't use the api to skip the auto-id generation
+        target = createContent("Supplier", id=target_id)
+        destination._setObject(target_id, target)
+        target = destination._getOb(target_id)
+
+    # Manually set the fields
+    # NOTE: always convert string values to unicode for dexterity fields!
+    target.title = api.safe_unicode(src_supplier.getName() or "")
+
+    move_reference_samples(src_supplier, target)
+
+    move_contacts(src_supplier, target)
+
+    # we set the fields with our custom setters
+    target.setRemarks(src_supplier.getRemarks())
+    target.setWebsite(src_supplier.getWebsite())
+    target.setNIB(src_supplier.getNIB())
+    target.setIBAN(src_supplier.getIBN())
+    target.setSwiftCode(src_supplier.getSWIFTcode())
+    target.setLabAccountNumber(src_supplier.getLabAccountNumber())
+    target.setTaxNumber(src_supplier.getTaxNumber())
+    target.setPhone(src_supplier.getPhone())
+    target.setFax(src_supplier.getFax())
+    target.setEmail(src_supplier.getEmailAddress())
+    target.setAccountType(src_supplier.getAccountType())
+    target.setAccountName(src_supplier.getAccountName())
+    target.setAccountNumber(src_supplier.getAccountNumber())
+    target.setBankName(src_supplier.getBankName())
+    target.setBankBranch(src_supplier.getBankBranch())
+    # copy address
+    target.setPostalAddress(src_supplier.getPostalAddress())
+    target.setPhysicalAddress(src_supplier.getPhysicalAddress())
+    target.setBillingAddress(src_supplier.getBillingAddress())
```

Comment:
I've just noticed that `state` from old address becomes now `subdivision1`. And `district` is stored under `subdivision2`. Otherwise, addresses will be partially migrated.

E.g.:

```python
postal_address = src_supplier.getPostalAddress()
if postal_address:
    postal_address["subdivision1"] = postal_address.pop("state", "")
    postal_address["subdivision2"] = postal_address.pop("district", "")
    target.setPostalAddress(postal_address)
```

Same for `PhysicalAddress` and for `BillingAddress`